### DISCUSSION TITLE
Control Keyboard from application

### DISCUSSION THREAD

**Discussion by mirij100** [AnySoftKeyboard/AnySoftKeyboard#2876] [Q&A]
Hello.

Is it possible to control this keyboard from application (on non rooted device), for example by settings myEditView.privateImeOptions = "options separated by ;"?
I need the following:

- switching keyboard layout (for example between Czech, Polish, Hungary etc.)
- prevent user from changing keyboard layout by long press Space key
- prevent keyboard floating mode
- hide (don't show) the upper line with icons
- set the y-position of the keyboard - I need to show my custom text/button under the keyboard

Regards, Jiří Mikeska

**Comment by mtx-z** [comment]
> prevent user from changing keyboard layout by long-press Space key

This is already in settings (Lang > Tips and plus) ("Astuce et plus" in french). You should triple-check that you've seen **all** settings pages, as there are options everywhere.

**Reply by mirij100** [reply]
But I need to control these from application programmatically. Not from UI.
